### PR TITLE
Add tests and fix bugs for declcfg server

### PR DESCRIPTION
**Pull request by joelanford**
Follow-up for #601 

Adds missing tests, and fixes a few bugs found by those tests :)

Signed-off-by: Joe Lanford <joe.lanford@gmail.com>

### CODE REVIEW

**Review by dinhxuanvu — APPROVED** [review]
Looking good. Approved.

File: pkg/api/conversion_test.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+	crdbackups  = `{"apiVersion":"apiextensions.k8s.io/v1beta1","kind":"CustomResourceDefinition","metadata":{"name":"etcdbackups.etcd.database.coreos.com"},"spec":{"group":"etcd.database.coreos.com","names":{"kind":"EtcdBackup","listKind":"EtcdBackupList","plural":"etcdbackups","singular":"etcdbackup"},"scope":"Namespaced","version":"v1beta2"}}`
```
Comment (anik120) [inline pkg/api/conversion_test.go:40]: the name here is `etcdbackups.etcd.database.coreos.com`

File: pkg/api/conversion_test.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+		Name:     "etcdoperator.v0.9.4",
+		Image:    "quay.io/operatorhubio/etcd:v0.9.4",
+		Replaces: "etcdoperator.v0.9.2",
+		Skips:    nil,
+		Properties: []property.Property{
+			property.MustBuildChannel("singlenamespace-alpha", "etcdoperator.v0.9.2"),
+			property.MustBuildPackage("etcd", "0.9.4"),
+			property.MustBuildPackageRequired("test", ">=1.2.3 <2.0.0-0"),
+			property.MustBuildGVKRequired("testapi.coreos.com", "v1", "Testapi"),
+			property.MustBuildGVK("etcd.database.coreos.com", "v1beta2", "EtcdBackup"),
```
Comment (anik120) [inline pkg/api/conversion_test.go:56]: But here it gets trimmed down to `etcd.database.coreos.com` 

Is it supposed to have the `etcdbackups` prefix? Or am I looking at the wrong place?

Comment (joelanford) [reply pkg/api/conversion_test.go:56]: Wrong place. This is the group portion of the GVK.

Comment (anik120) [reply pkg/api/conversion_test.go:56]: Ah, makes sense

**Comment by dinhxuanvu** [comment]
/hold
Waiting for Vu's PR to go first.

**Pull request merged**